cpp/src/text/ngrams_tokenize.cu

```diff
@@ -158,21 +158,21 @@ std::unique_ptr<cudf::column> ngrams_tokenize(
 
   // first, get the number of tokens per string to get the token-offsets
   // Ex. token-counts = [3,2]; token-offsets = [0,3,5]
-  rmm::device_vector<int32_t> token_offsets(strings_count + 1);
-  auto d_token_offsets = token_offsets.data().get();
+  rmm::device_uvector<int32_t> token_offsets(strings_count + 1, stream);
+  auto d_token_offsets = token_offsets.data();
   thrust::transform_inclusive_scan(rmm::exec_policy(stream),
                                    thrust::make_counting_iterator<cudf::size_type>(0),
                                    thrust::make_counting_iterator<cudf::size_type>(strings_count),
                                    d_token_offsets + 1,
                                    strings_tokenizer{d_strings, d_delimiter},
                                    thrust::plus<int32_t>());
-  CUDA_TRY(cudaMemsetAsync(d_token_offsets, 0, sizeof(int32_t), stream.value()));
-  auto total_tokens = token_offsets[strings_count];  // Ex. 5 tokens
+  token_offsets.set_element(0, 0, stream);
```

Comment:
This should be made async, but we have to be careful because it's not technically safe to use a literal as input to the async version. I will be filing a PR to remove the ability to do so. 

```suggestion
  auto init = 0;
  token_offsets.set_element_async(0, init, stream);
```

Reply:
> This should be made async, but we have to be careful because it's not technically safe to use a literal as input to the async version. I will be filing a PR to remove the ability to do so.

Does that mean I need to do a sync at the end of the function since the `init` variable will go out of scope then?

Reply:
Actually `back_element()` does a sync so I should be good if this occurs after the last `set_element_async()` call in the same function.

Reply:
Good observation. Yes back_element should be enough.